Nardri/trisixty-buys-API
# #162832764 Verify email on user registration

#### What does this PR do?
This PR implements an email verification on user registration.

#### Description of Task to be completed?
As a developer, I should be able to send out a clickable verify token to the registered email by making a `POST` request with the valid details to `/user/register`.

#### How should this be manually tested?
- Get the branch with `git fetch origin ft-email-verification-162832764`
- Checkout to the branch with `git checkout  ft-email-verification-162832764`
- Activate the env with `pipenv shell` and install dependencies with `pipenv sync`
- Run test `pytest`
- Run the application with `flask run`
- Make a POST request to`api/v1/user/register`

REQUEST
`POST /user/register`
```
{
    "firstname": "test",
    "lastname": "user",
    "email": "test.user@example.com",
    "password": "password1234"
}
```

Response data should have the following structure
```
{
    "message" : "User successfully registered, please check your mail to verify your account.",
    "status" : "success"
}
```
#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#162832764](https://www.pivotaltracker.com/story/show/162832764)

#### Screenshots (if appropriate)

#### Questions:
N/A
